x-pack/test/cloud_security_posture_functional/agentless/create_agent.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CLOUD_CREDENTIALS_PACKAGE_VERSION } from '@kbn/cloud-security-posture-plugin/common/constants';
+import * as http from 'http';
+import expect from '@kbn/expect';
+import type { FtrProviderContext } from '../ftr_provider_context';
+import { setupMockServer } from './mock_agentless_api';
+// eslint-disable-next-line import/no-default-export
+export default function ({ getPageObjects, getService }: FtrProviderContext) {
+  const mockAgentlessApiService = setupMockServer();
+  const pageObjects = getPageObjects([
+    'common',
+    'cspSecurity',
+    'security',
+    'header',
+    'cisAddIntegration',
+  ]);
+
+  const CIS_AWS_OPTION_TEST_ID = 'cisAwsTestId';
+
+  const AWS_SINGLE_ACCOUNT_TEST_ID = 'awsSingleTestId';
+
+  describe('Agentless cloud', function () {
+    let cisIntegration: typeof pageObjects.cisAddIntegration;
+    let mockApiServer: http.Server;
+
+    before(async () => {
+      cisIntegration = pageObjects.cisAddIntegration;
+      mockApiServer = await mockAgentlessApiService.listen(8089); // Start the usage api mock server on port 8081
+    });
+
+    after(async () => {
+      await await pageObjects.cspSecurity.logout();
+      mockApiServer.close();
+    });
+
+    it(`should create agentless-agent`, async () => {
+      const integrationPolicyName = `cloud_security_posture-${new Date().toISOString()}`;
+      await cisIntegration.navigateToAddIntegrationCspmWithVersionPage(
+        CLOUD_CREDENTIALS_PACKAGE_VERSION
+      );
+
+      await cisIntegration.clickOptionButton(CIS_AWS_OPTION_TEST_ID);
+      await cisIntegration.clickOptionButton(AWS_SINGLE_ACCOUNT_TEST_ID);
+
+      await cisIntegration.inputIntegrationName(integrationPolicyName);
+
+      await cisIntegration.selectSetupTechnology('agentless');
+      await cisIntegration.selectAwsCredentials('direct');
+
+      await pageObjects.header.waitUntilLoadingHasFinished();
+
+      await cisIntegration.clickSaveButton();
+      await pageObjects.header.waitUntilLoadingHasFinished();
+
+      expect(
+        (await cisIntegration.cisAws.getPostInstallCloudFormationModal()) ? true : false
```

Comment:
Agentless integrations won't show the CloudFormation modal for the happy flow.   I don't think we need to check the assertion for Cloudformation Modal since this feature is for Agent-based Integrations. If this test case for creating agentless agent,  as E2E  user journey test case, the user will do the following:

1. Save Agentless integration successfully, 
2. Checks if successful toast message to ensure no api error were thrown and agentless agent created was successfully
3. Navigates agentless integration policies tab in the list 
4. See if newly agentless integration was created with agent count of 1?

Reply:
You are not wrong,  although I have never seen a successful toast message. 🤔 
I know that the onboarding flow will change for Agentless and I was thinking that you would build on this set of FTRs.

I originally did have the tests navigate to the integration policy page and validate that the policy names match what we expected but I was getting some flaky-ness.  I will add it back in a see if it is still flakey.

Reply:
Ahh okay interesting 
usually the [successful toast message](https://github.com/elastic/kibana/blob/83e78ea5da58c34dafe10d0c107859b678a8f9e7/x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/single_page_layout/hooks/form.tsx#L439) will show but maybe since we are using MSW server then we bypass the notification.

Reply:
I just pushed changes to navigate to the CSPM integration page and check the policy and agent exist.